include/hse/hse.h

```diff
@@ -754,6 +754,11 @@ hse_kvs_put(
  * so that only one thread is actively performing an operation in the context of
  * a particular transaction at any particular time.
  *
+ * TODO: Add detail on thread safety, such as: client must serialize all
```

Comment:
Alex what were you thinking here? I am happy to make a PR to add something here, but I am not fully grasping what you are describing.